tests/unit/s2n_server_hello_test.c

```diff
@@ -154,6 +155,51 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(server_conn));
     }
 
+    /* Test Server Hello Recv with invalid cipher */
+    {
+        struct s2n_config *server_config;
+        struct s2n_config *client_config;
+
+        struct s2n_connection *server_conn;
+        struct s2n_connection *client_conn;
+
+        EXPECT_NOT_NULL(server_config = s2n_config_new());
+        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, server_config));
+
+        EXPECT_NOT_NULL(client_config = s2n_config_new());
+        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, client_config));
+
+        struct s2n_stuffer *server_stuffer = &server_conn->handshake.io;
+
+        const uint32_t total = S2N_TLS_PROTOCOL_VERSION_LEN
+            + S2N_TLS_RANDOM_DATA_LEN
+            + SESSION_ID_SIZE
+            + server_conn->session_id_len
+            + S2N_TLS_CIPHER_SUITE_LEN
+            + COMPRESSION_METHOD_SIZE;
```

Comment:
Does this test need the size check, since you're not testing a size? Shouldn't "not zero" be sufficient here?

Reply:
It doesn't need the size check (removed). It does need to know the amount of data to copy from stuffer to stuffer.

Reply:
Nite: Instead of calculating the size (which could change, and isn't relevant to this test), I'd just use s2n_stuffer_data_available.

Reply:
Updated. That check is also removed from other tests where it is irrelevant to the specific test.